src/de.rs

```diff
@@ -1358,3 +1344,78 @@ where
         self.map.next_value_seed(seed)
     }
 }
+
+/// Deserialize a DAG-CBOR encoded CID.
+///
+/// This is without the CBOR tag information. It is only the CBOR byte string identifier (major
+/// type 2), the number of bytes, and a null byte prefixed CID.
+///
+/// The reason for not including the CBOR tag information is the [`Value`] implementation. That one
+/// starts to parse the bytes, before we could interfere. If the data only includes a CID, we are
+/// parsing over the tag to determine whether it is a CID or not and go from there.
+struct CidDeserializer<'a, R>(&'a mut Deserializer<R>);
+
+//impl<'de, 'a: 'de, R> de::Deserializer<'de> for &'a mut CidDeserializer<'a, R>
+impl<'de, 'a, R> de::Deserializer<'de> for &'a mut CidDeserializer<'a, R>
+where
+    R: Read<'de>,
+{
+    type Error = Error;
+
+    fn deserialize_any<V: de::Visitor<'de>>(self, _visitor: V) -> Result<V::Value> {
+        unreachable!()
+    }
+
+    #[inline]
+    fn deserialize_bytes<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
+        // Match on the major type, it must be a byte string (major type 2)
+        let len = match self.0.parse_u8()? {
```

Comment:
Can we not reuse our vec deserialization code?

Reply:
I mean, we already have logic for deserializing byte buffers, is there any way to reuse that here? (not critical, just a thought)

Reply:
Oh I see. yes we have, but it's not really worth it (given the current code structure of this library).